cdatgui/editors/widgets/legend_widget.py

```diff
@@ -0,0 +1,552 @@
+from types import FunctionType
+
+from cdatgui.editors.widgets.dict_editor import DictEditorWidget
+from PySide import QtCore, QtGui
+from cdatgui.editors.model import legend
+from cdatgui.editors.preview.legend_preview import LegendPreviewWidget
+from cdatgui.bases.window_widget import BaseOkWindowWidget
+from cdatgui.editors.colormap import QColormapEditor
+from cdatgui.utils import pattern_thumbnail
+from cdatgui.bases.dynamic_grid_layout import DynamicGridLayout
+from functools import partial
+
+
+class CustomScrollArea(QtGui.QScrollArea):
+    def resizeEvent(self, ev):
+        super(CustomScrollArea, self).resizeEvent(ev)
+        self.widget().setGeometry(0, 0, self.width(), self.height())
+
+
+class CustomFillWidget(QtGui.QWidget):
+    colorChanged = QtCore.Signal(int, int)
+    opacityChanged = QtCore.Signal(int, int)
+    buttonColorChanged = QtCore.Signal(QtGui.QPushButton, int)
+    patternChanged = QtCore.Signal(int, int)
+    attributeChanged = QtCore.Signal()
+    colormapChanged = QtCore.Signal(str)
+    createColormapRequest = QtCore.Signal(FunctionType)
+
+    def __init__(self, index, label, parent=None):
+        super(CustomFillWidget, self).__init__()
+
+        # Variables
+        self.index = index
+        self.label = label
+        self.color = None
+        self.pattern = 1
+
+        # Label
+        level_layout = QtGui.QHBoxLayout()
+        level_layout.addWidget(QtGui.QLabel(label))
+
+        # Color button
+        self.color_button = QtGui.QPushButton()
+
+        self.color_button.setFixedSize(100, 40)
+        self.color_button.clicked.connect(lambda: self.createColormapRequest.emit(self.changeColor))
```

Comment:
If the only thing ever emitting createColormapRequests is passing the same method of this object, couldn't this signal not have an argument?